ext/auto_flush.c

```diff
@@ -38,6 +60,36 @@ ZEND_RESULT_CODE ddtrace_flush_tracer(bool force_on_startup, bool collect_cycles
         return SUCCESS;
     }
 
+    if (get_DD_EXPERIMENTAL_APPSEC_STANDALONE_ENABLED()) {
+        if (!DDTRACE_G(asm_event_emitted) && !trace_contains_appsec_event(&trace) && !ddtrace_standalone_limiter_allow()) {
+            if (Z_TYPE(trace) != IS_ARRAY) {
+                LOG(ERROR, "Unexpected trace type. Dropping trace");
+                return SUCCESS;
+            }
+
+            zval *root_span = zend_hash_index_find(Z_ARR(trace), 0);
+            if (!root_span || Z_TYPE_P(root_span) != IS_ARRAY) {
+                LOG(ERROR, "Root span not found. Dropping trace");
+                return SUCCESS;
+            }
+
+            zval *metrics = zend_hash_str_find(Z_ARR_P(root_span), ZEND_STRL("metrics"));
+            if (!metrics || Z_TYPE_P(metrics) != IS_ARRAY) {
+                LOG(ERROR, "Metrics not found. Dropping trace");
+                return SUCCESS;
+            }
+
+            zval *sampling_priority = zend_hash_str_find(Z_ARR_P(metrics), ZEND_STRL("_sampling_priority_v1"));
+            if (!sampling_priority || (Z_TYPE_P(sampling_priority) != IS_DOUBLE && Z_TYPE_P(sampling_priority) != IS_LONG)) {
+                LOG(ERROR, "Invalid sampling priority. Dropping trace");
+                return SUCCESS;
+            }
+            ZVAL_LONG(sampling_priority, PRIORITY_SAMPLING_AUTO_REJECT);
```

Comment:
No, we need to set them to auto reject

Reply:
How do you reconcile that statement with the text I quoted? Because in addition to what I quoted, the spec also says:

> Other sources of force-keep sampling priorities, such as errors, crucial to ASM’s backend side monitoring, are still allowed in standalone ASM mode.

So it seems to me that force_keep traces should not be rejected. If my understanding is wrong, please explain.

Reply:
This is the scenario when apm should behave as disable. There is no asm events, no appsec upstream services and it's not used as a heartbeat. In that case we need to let the trace drop. Does it makes sense?